api/konnect/v1alpha1/konnect_extension_types.go

```diff
@@ -180,20 +181,50 @@ type KonnectExtensionStatus struct {
 	// +kubebuilder:validation:Optional
 	DataPlaneClientAuth *DataPlaneClientAuthStatus `json:"dataPlaneClientAuth,omitempty"`
 
+	// Konnect contains the status information related to the Konnect Control Plane.
+	//
+	// +kubebuilder:validation:Optional
+	Konnect *KonnectExtensionControlPlaneStatus `json:"konnect,omitempty"`
+
 	// Conditions describe the current conditions of the KonnectExtensionStatus.
 	// Known condition types are:
 	//
-	// * "Accepted"
-	// * "SecretProvisioned"
-	// * "ControlPlaneRefValid"
-	//
 	// +optional
 	// +listType=map
 	// +listMapKey=type
 	// +kubebuilder:validation:MaxItems=8
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 }
 
+// ClusterType is the type of the Konnect Control Plane.
+type ClusterType string
+
+const (
+	// ClusterTypeControlPlane is the type of the Konnect Control Plane.
+	ClusterTypeControlPlane ClusterType = "ControlPlane"
+	// ClusterTypeK8sIngressController is the type of the Kubernetes Control Plane.
+	ClusterTypeK8sIngressController ClusterType = "K8SIngressController"
+)
+
+// KonnectExtensionControlPlaneStatus contains the Konnect Control Plane status information.
+type KonnectExtensionControlPlaneStatus struct {
+	// ControlPlaneID is the Konnect ID of the ControlPlane this KonnectExtension is associated with.
+	//
+	// +kubebuilder:validation:Required
+	ControlPlaneID string `json:"controlPlaneID,omitempty"`
```

Comment:
`omitempty` shouldn't be used with required fields (ref: https://github.com/JoelSpeed/kal?tab=readme-ov-file#requiredfields)

Reply:
yep yep good catch, that's probably a leftover